src/error.rs

```diff
@@ -0,0 +1,33 @@
+use defines::AfError;
+
+pub type ErrorCallback = Fn(AfError);
+
+pub static mut HANDLE_ERROR: &'static ErrorCallback = &handle_error_general; 
+
+pub fn register_error_handler(callback: &'static ErrorCallback) {
+    unsafe {
+        HANDLE_ERROR = callback;
+    }
+}
+
+pub fn handle_error_general(error_code: AfError) {
+    match error_code {
+        AfError::SUCCESS            => {}, /* No-op */
```

Comment:
Also you would need this: https://doc.rust-lang.org/std/sync/struct.StaticMutex.html as static does not implement `Drop`